### PR TITLE
Fix some comments

### PR DESCRIPTION
**Pull request by wenovus**
Fix outdated design doc comment and a couple other comments.

### CODE REVIEW

File: ygen/gogen.go [hunk 1]
```diff
@@ -1948,7 +1948,10 @@ func findMapPaths(parent *Directory, fieldName string, compressPaths, absolutePa
 			return nil, fmt.Errorf("invalid compressed schema, could not find the key %s or the grandparent of %s", k.Name, k.Path())
 		}
 
-		if reflect.DeepEqual(util.SchemaPathNoChoiceCase(k), fieldSlicePath) && k.Parent.Parent.Dir[k.Name].Type.Kind == yang.Yleafref {
+		// If a key of the list is a leafref to the field (goyang makes
+		// leafrefs have the same parent as the entry to which they
+		// point), then add this as an alternative path to the field.
+		if cmp.Equal(util.SchemaPathNoChoiceCase(k), fieldSlicePath) && k.Parent.Parent.Dir[k.Name].Type.Kind == yang.Yleafref {
```
Comment (wenovus) [inline ygen/gogen.go]: Sorry this comment is wrong, I need to fix it.

File: genutil/common.go [hunk 2]
```diff
@@ -104,15 +104,19 @@ type CompressBehaviour int64
 // No dimension spans across all others' options, so can't extract any out
 // without having to error out for some combinations.
 const (
-	// Uncompressed means to not compress the schema.
+	// Uncompressed means to not compress the generated code relative to
```
Comment (progressnerd) [inline genutil/common.go]: I think they could each be phrased in an active voice:
"means to" -> "does"

Comment (progressnerd) [inline genutil/common.go]: drop "relative to the schema" because it doesn't add anything?

File: genutil/common.go [hunk 3]
```diff
@@ -104,15 +104,19 @@ type CompressBehaviour int64
...
+	// the schema.
 	Uncompressed CompressBehaviour = iota
-	// UncompressedExcludeDerivedState excludes config false subtrees.
+	// UncompressedExcludeDerivedState excludes config false subtrees in
+	// the generated code.
 	UncompressedExcludeDerivedState
-	// PreferIntendedConfig indicates to use the "config" version of a
-	// schema entry instead of its "state" counterpart when both exist.
+	// PreferIntendedConfig indicates to generate only the "config" version
```
Comment (progressnerd) [inline genutil/common.go]: "indicates to generate" -> "generates"

File: genutil/common.go [hunk 4]
```diff
@@ -104,15 +104,19 @@ type CompressBehaviour int64
...
+	// of a field when it exists under both "config" and "state" containers
```
Comment (progressnerd) [inline genutil/common.go]: this tells me what happens when it exists under both, but what happens when it exists under one and not the other?

File: genutil/common.go [hunk 5]
```diff
@@ -104,15 +104,19 @@ type CompressBehaviour int64
...
+	// of its parent YANG model.
 	PreferIntendedConfig
-	// PreferOperationalState indicates to use the "state" version of a
-	// schema entry instead of its "config" counterpart when both exist.
+	// PreferOperationalState indicates to generate only the "state"
```
Comment (progressnerd) [inline genutil/common.go]: same as above: "indicates to generate" -> "generates" and what happens when a field exists under one of either "config" or "state"

File: genutil/common.go [hunk 6]
```diff
@@ -104,15 +104,19 @@ type CompressBehaviour int64
...
+	// version of a field when it exists under both "config" and "state"
+	// containers of its parent YANG model.
 	PreferOperationalState // prefer applied config
```
Comment (progressnerd) [inline genutil/common.go]: drop "prefer applied config"

File: genutil/common.go [hunk 7]
```diff
@@ -104,20 +104,25 @@ type CompressBehaviour int64
...
 	Uncompressed CompressBehaviour = iota
 	// UncompressedExcludeDerivedState excludes config false subtrees in
 	// the generated code.
 	UncompressedExcludeDerivedState
-	// PreferIntendedConfig indicates to generate only the "config" version
-	// of a field when it exists under both "config" and "state" containers
-	// of its parent YANG model.
+	// PreferIntendedConfig generates only the "config" version of a field
+	// when it exists under both "config" and "state" containers of its
+	// parent YANG model. This is done to resolve field conflicts that
```
Comment (progressnerd) [inline genutil/common.go]: I don't get the "This is done ..." sentence. Maybe just drop it?

**Pull request merged**